trace/client.go

```diff
@@ -69,12 +86,33 @@ func (c *Client) run(ctx context.Context) {
 	if c.report != nil {
 		go c.report(ctx)
 	}
+
+	for _, b := range c.flushBackends {
+		go runFlushableBackend(ctx, c.records, b.backend, b.notify)
+	}
+}
+
+func runFlushableBackend(ctx context.Context, spans chan *sendOp, backend ClientBackend, flushNotify chan chan<- error) {
+	defer backend.Close()
+
 	for {
```

Comment:
Good question - it's fine to check for `spans` closedness (`.Close` cancels the context and closes the chan), but I'll make it check for context cancelation. Seems cleaner.

Reply:
hmmm...actually, now that i'm looking at the fix, i'm not sure.

If the `spans` channel is also getting closed by the same action that initiates context cancelation, then it might be preferable to do that. Otherwise, depending on (crazy scheduler ordering), i suspect it may be possible that there would be at least one pass through the select loop that hits the first case, resulting in a SendSync with a default value op.span. That might result in generating some garbage.

Reply:
Hah! I'm reasonably certain that can't happen because we pass the client's context along into SendSync - backend implementations can (ours certainly do) check if the context is canceled & abort early.

Reply:
Also, I think this helps us avoid race conditions since if a Record event is accepted onto the send channel, that means something might wait for the result, and it's better that they get a "context canceled" error than nothing at all

Reply:
ok, SGTM then 😄